app/models/feature.rb

```diff
@@ -1,170 +1,54 @@
-class Feature < ActiveRecord::Base
-  belongs_to :spatial_model, :polymorphic => :true, :autosave => false
+class Feature < AbstractFeature
+  class_attribute :automatically_refresh_aggregate
+  self.automatically_refresh_aggregate = true
 
-  attr_writer :make_valid
+  class_attribute :lowres_precision
+  self.lowres_precision = 5
 
-  FEATURE_TYPES = %w(polygon point line)
+  has_one :aggregate_feature, lambda { |feature| aggregated.where(:spatial_model_type => feature.spatial_model_type) }, :foreign_key => :spatial_model_id, :primary_key => :spatial_model_id
 
-  before_validation :sanitize_feature_type
-  validates_presence_of :geog
-  validate :validate_geometry
   validates_inclusion_of :feature_type, :in => FEATURE_TYPES
-  before_save :sanitize
-  after_save :cache_derivatives
 
-  def self.cache_key
-    "#{maximum(:id)}-#{count}"
-  end
+  after_save :refresh_aggregate, if: :automatically_refresh_aggregate?
 
-  def self.with_metadata(k, v)
-    if k.present? && v.present?
-      where('metadata->? = ?', k, v)
-    else
-      all
-    end
-  end
+  def self.defer_aggregate_refresh(&block)
```

Comment:
It seems kind of weird to me for the `Feature` to handle the aggregate refresh. Seems more like something that the spatial model should handle?

Reply:
Let's leave it as is for now and see how it works in practice.